ecosystem/wallet-apps/tonkeeper.mdx

```diff
@@ -2,4 +2,97 @@
 title: "Tonkeeper"
 ---
 
-Stub
+[Tonkeeper](https://tonkeeper.com/) is a non-custodial wallet for the TON blockchain. It supports TON coins, jettons, NFTs, TON Connect, and is available as mobile apps, desktop applications, and browser extensions.
+
+<img
+  src="/resources/images/tonkeeper/Tonkeeper.png"
+  alt="Main dashboard interface"
+  height="300"
+  className="rounded-lg"
+/>
+
+## Key features
+
+- **TON Connect integration:** integrates with dApps through the TON Connect protocol, allowing users to securely connect their wallet to web-based applications
+- **Battery payments:** provides a prepaid balance that covers transaction fees (gas) when users don't have enough Toncoin in their account
+- **W5 wallet implementation:** supports W5, the latest wallet smart contract standard for the TON blockchain
+- **Built-in dApp browser:** provides access to decentralized applications (dApps) directly within the wallet
+- **Multi-asset support:** supports digital assets including TON coins, jettons and NFTs
+- **Cross-platform architecture:** runs on web, desktop, and mobile environments
+
+### Supported platforms
+
+- **Mobile apps:** iOS, Android
+- **Desktop apps:** Windows, macOS, Linux
+- **Browser extensions:** Chrome, Firefox, Brave, Yandex, Edge
+- **Web version:** runs directly in the browser, no installation required
+- **Telegram:** all known [Telegram clients](https://telegram.org/apps)
+
+## Create a wallet
+
+To create a new wallet:
+
+1. Open Tonkeeper
+1. Select **Create new wallet**
+1. Set a PIN code for transaction confirmation
+1. Add a name for your wallet
+1. Back up your recovery phrase (24-word mnemonic)
+
+<Tip>The recovery phrase is the only way to restore access to your wallet if the app is deleted or your device is lost. Make sure to save it in a secure place.</Tip>
+
+<img
+  src="/resources/images/tonkeeper/backup.png"
+  alt="Backup recovery phrase"
+  height="50%"
+  className="rounded-lg"
+/>
+
+### For Testnet
+
+Tonkeeper requires a mainnet wallet before creating a testnet account. The testnet account derives from your existing wallet's mnemonic.
+
+To add a testnet account:
+
+1. From the main screen, tap your **wallet name**
+1. Select **Add Wallet → Testnet Account**
+1. Enter your 24-word mnemonic
+
+<img
+  src="/resources/images/tonkeeper/testnet-account.jpg"
+  alt="Add Testnet Account"
+  height="300"
+  className="rounded-lg"
+/>
+
+## Deploy wallet
+
+Before you can use your wallet on-chain to send transactions or interact with smart contracts, you need to deploy it.
```

Comment:
this needs to be explained more carefully, including the version of your wallet contract